### DISCUSSION TITLE
File Save Location (Attachment Location)

### DISCUSSION THREAD

**Discussion by mahfuzplabon** [chatwoot/chatwoot#2521] [Q&A] [answered]
Hello,

I followed the Docker tutorial, everything is running smooth. I want to know where the images/other files which have been uploaded by the visitor is saved ?? And is it deleted after some day ? Or, usually never deleted ?? Can I manually remove those ??

Right Now, I'm seeing this location:

https://URL/rails/active_storage/disk/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdDRG9JYTJWNVNTSWhPRGN3TURWa09EZHZOblZwZG5wMk5tWjNPREF4Tm5OeVkyeGhNZ1k2QmtWVU9oQmthWE53YjNOcGRHbHZia2tpT1dsdWJHbHVaVHNnWm1sc1pXNWhiV1U5SWs1SlJDNVFUa2NpT3lCbWFXeGxibUZ0WlNvOVZWUkdMVGduSjA1SlJDNVFUa2NHT3daVU9oRmpiMjUwWlc1MFgzUjVjR1ZKSWc1cGJXRm5aUzl3Ym1jR093WlUiLCJleHAiOiIyMDIxLTA2LTI3VDIwOjM1OjI1LjYyMVoiLCJwdXIiOiJibG9iX2tleSJ9fQ==--c5e77960c80e0b0bd13ce0b3811303bfb9f1f77d/NID.PNG?content_type=image%2Fpng&disposition=inline%3B+filename%3D%22NID.PNG%22%3B+filename%2A%3DUTF-8%27%27NID.PNG

**Comment by pranavrajs** [comment, marked as answer]
@mahfuzplabon If you are using Docker based installation, you need to configure one of the cloud storage services, otherwise the files would be deleted the next time the container is rebooted. 

Please see this document: https://www.chatwoot.com/docs/self-hosted/deployment/storage/supported-providers

**Reply by Toninobandolero** [reply]
Hi! 
is there any posibility to save the images locally using docker container in CapRover?
adding public/uploads on persistance folders??

I´m using ACTIVE_STORAGE_SERVICE=local
but images doesn´t stay when i restart 



If it is NOT possible, can somebody explain me how to complete this?:

Using Google GCS
Configure the following env variables.

ACTIVE_STORAGE_SERVICE=google
GCS_PROJECT= 
GCS_CREDENTIALS=
GCS_BUCKET=

the value of the GCS_CREDENTIALS should be a json formatted string containing the following keys

{
  "type": "service_account",
  "project_id" : "",
  "private_key_id" : "",
  "private_key" : "",
  "client_email" : "",
  "client_id" : "",
  "auth_uri" : "",
  "token_uri" : "",
  "auth_provider_x509_cert_url" : "",
  "client_x509_cert_url" : ""
}


Thanks!
